contents/projectManagement/githubActions.md

```diff
@@ -0,0 +1,162 @@
+
+#### Action
+An action is a piece of code that performs one single modular task. For example, [checking out a repository](https://github.com/actions/checkout).
+
+#### Triggers
+Workflows can be triggered by events on the Github repository, ranging from opening pull requests or issues, making commits to scheduled events, or even from an external event by calling Github's REST API endpoint.
+
+
+### How is Github Actions being used
+
+#### Simple CI/CD workflow example for a NodeJS app
+1. Trigger when a commit/pull request is made to master
+2. Build
+3. Run tests on ubuntu
+4. Deploy app if tests pass
+
+##### [Workflow file](https://github.com/actions/starter-workflows/blob/master/ci/node.js.yml)
+```yaml
+name: Node.js CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x]
+
+    steps:
+    - run: echo Hello World!
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
+    - run: npm run build --if-present
+    - run: npm test
+      env:
+        CI: true
+    - run: npm run deploy
+```
+
+
+
+## How to Get Started with X?
+<!--
+Provide a learning path for the reader. Try to give one good learning path rather than many random resources.
+-->
+The [official documentation](https://help.github.com/en/actions/getting-started-with-github-actions/overview) provides a useful overview of Github Actions (summarized in this article), but it does not give you a clear and easy to follow path to getting started. This is a mini-guide for you to get started with Github Actions.
+
+### Set up a simple workflow (Node.js)
+Fortunately for us, Github has made it extremely easy to set up a simple workflow without even leaving their website. 
+
+![](https://i.imgur.com/zC4ScYn.png)
+
+1. Head to the Actions tab on your repository
+2. You can choose to set up a workflow from a template or from scratch. We will use the Node.js template in this example
+3. The online editor is very powerful and easy to use, boasting features such as inline documentation and autocomplete
+![](https://i.imgur.com/gknkruJ.png)
+4. The template workflow (similar to the example above) will
+    1. Begin when a commit is pushed or a pull request is opened to `master`
+    2. Checkout the current repo (using `actions/checkout`)
+    3. Install dependencies `npm ci`
+    4. Build `npm run build --if-present`
+    5. Run tests `npm run test`
+    6. Report the status of the workflow via a status badge
+    ![](https://i.imgur.com/3Tsm2Ii.png)
+5. Since we set the workflow up on the `master` branch, when we commit the newly-created workflow file, we can see Github Actions in action by navigating to the Actions tab
+    ![](https://i.imgur.com/ZxBlv8d.png)
+
+
+
+
+## Where to Go from Here?
```

Comment:
Probably you may recommend some widely used actions like [cache](https://github.com/actions/cache), etc.